lib/game_client/ogs/ogs_game.dart

```diff
@@ -122,6 +130,11 @@ class OGSGame extends Game {
         return;
       }
 
+      if (point == (-1, -1)) {
```

Comment:
tbh, I'm surprised this needs special handling, given that the implementation of `toSGF` treats (-1, -1) as pass:

https://github.com/ale64bit/WeiqiHub/blob/e950f13578c1db63834455bd30f5451d59ae91e1/lib/wq/wq.dart#L28

IMO, passes should be treated as moves and added to the game tree, though I'm not sure it matters in practice.

Reply:
Yeah, this is due to some internal weirdness. The board object doesn't have a concept of "pass", it just allows you to play moves. The game tree object, however, does add pass as a separate move. If Dart had union-types, this could be expressed a bit better I suppose.

Reply:
I guess I don't mind the existing type, more that we have multiple ways to represent "pass".  In `Game.moves()` pass must be represented as `null`, whereas in SGF methods, pass is represented as `Point(-1, -1)`.  I believe it would be simple for `GamePage` to handle the `Point` representation instead:

```diff
    widget.game.moves().forEach((mv) {
-     if (mv == null) {
+     if (mv.point == (-1, -1)) 
+     // another issue here is that onPass does not update GamePage._gameTree
+     // which might cause issues if you add a features that require a complete tree
        onPass();
        widget.gameListener?.onPass(widget.game.id);
      } else {
        onMove(mv);
        widget.gameListener?.onMove(widget.game.id, mv);
      }
    });
```